tests/integration.rs

```diff
@@ -413,20 +424,36 @@ fn noxattrs(#[case] harness: Harness) {
 	let file = File::open(d.path().join("file1")).unwrap();
 	let xattrs = file.list_xattr().unwrap().collect::<Vec<_>>();
 	assert_eq!(xattrs.len(), 0);
+}
 
-	freebsd! {
-		let num = unsafe {
-			libc::extattr_list_fd(file.as_raw_fd(), EXTATTR_NAMESPACE_USER, std::ptr::null_mut(), 0)
-		};
-		assert_eq!(num, 0);
-
-		let name = cstr!(b"test");
-		let num = unsafe {
-			libc::extattr_get_fd(file.as_raw_fd(), EXTATTR_NAMESPACE_USER, name.as_ptr(), std::ptr::null_mut(), 0)
-		};
-		assert_eq!(num, -1);
-		assert_eq!(errno(), libc::ENOATTR);
-	}
+#[cfg(target_os = "freebsd")]
+#[apply(all_images)]
+fn noxattrs_freebsd(#[case] harness: Harness) {
```

Comment:
And here:
```suggestion
fn noxattrs(#[case] harness: Harness) {
```

Reply:
This test isn't really FreeBSD-specific, either.  And it's testing two things.  It should be two separate test cases: one for extattr_list_fd size with no attrs, and one for extattr_get_fd size with a nonexistent attribute.